Formula/helm-diff.rb

```diff
@@ -0,0 +1,32 @@
+class HelmDiff < Formula
+  desc "Helm plugin shows a diff explaing what a helm upgrade would change"
+  homepage "https://github.com/databus23/helm-diff"
+  url "https://github.com/databus23/helm-diff.git",
+    :tag      => "v2.11.0+5",
+    :revision => "33ee5a32d005db46d9efefeecb97d7b9aa125428"
+
+  depends_on "glide" => :build
+  depends_on "go" => :build
+  depends_on "kubernetes-cli"
+  depends_on "kubernetes-helm"
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"
+
+    dir = buildpath/"src/github.com/databus23/helm-diff"
+    dir.install buildpath.children
+
+    cd dir do
+      system "glide", "install"
+      system "go", "build", "-ldflags", "-s -w -X github.com/databus23/helm-diff/cmd.Version=#{version}",
+        "-o", bin/"diff"
+
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/diff version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.